src/vm/hints/find_element_hint.rs

```diff
@@ -165,21 +165,136 @@ pub fn find_element(
     }
 }
 
+pub fn search_sorted_lower(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let array_ptr_addr = get_address_from_var_name("array_ptr", &ids, vm, hint_ap_tracking)?;
+    let elm_size_addr = get_address_from_var_name("elm_size", &ids, vm, hint_ap_tracking)?;
+    let n_elms_addr = get_address_from_var_name("n_elms", &ids, vm, hint_ap_tracking)?;
+    let index_addr = get_address_from_var_name("index", &ids, vm, hint_ap_tracking)?;
+    let key_addr = get_address_from_var_name("key", &ids, vm, hint_ap_tracking)?;
+
+    match (
+        vm.memory.get(&array_ptr_addr),
+        vm.memory.get(&elm_size_addr),
+        vm.memory.get(&n_elms_addr),
+        vm.memory.get(&index_addr),
+        vm.memory.get(&key_addr),
+    ) {
+        (
+            Ok(Some(maybe_rel_array_ptr)),
+            Ok(Some(maybe_rel_elm_size)),
+            Ok(Some(maybe_rel_n_elms)),
+            Ok(_),
+            Ok(Some(maybe_rel_key)),
+        ) => {
```

Comment:
You can probably replace most of this with the new `get_integer_from_var_name` and `get_relocatable_from_var_name` helper functions.
There's also a helper to get the range check builtin as well.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/305/commits/c3d02a7a40339b3dfb695aaea764d78c5fe4f589